application/app/controllers/dataverse/datasets_controller.rb

```diff
@@ -68,19 +70,22 @@ def init_project_service
 
   def find_dataset_by_persistent_id
     @persistent_id = params[:persistent_id]
+    @version = params[:version]
     begin
-      @dataset = @service.find_dataset_version_by_persistent_id(@persistent_id)
+      @dataset = @service.find_dataset_version_by_persistent_id(@persistent_id, version: @version)
       unless @dataset
-        log_error('Dataset not found.', { dataverse: @dataverse_url, persistent_id: @persistent_id })
-        redirect_back_to_app(alert: t("dataverse.datasets.dataset_not_found", dataverse_url: @dataverse_url, persistent_id: @persistent_id))
+        log_error('Dataset not found.', { dataverse: @dataverse_url, persistent_id: @persistent_id, version: @version })
+        redirect_back_to_app(alert: t("dataverse.datasets.dataset_not_found", dataverse_url: @dataverse_url, persistent_id: @persistent_id, version: @version))
         return
       end
+      # SET VERSION IN CASE NOT SET IN REQUEST
+      @version = @dataset.version
```

Comment:
shouldn't this assignment be conditional instead of always override?

Reply:
It is safe to always override and it simplifies the logic.
As well, the dataset version even if different due to a bug or something else, should be the one used across the app. Otherwise we will show data for version 1 and the version badge will show something else.

Reply:
ok